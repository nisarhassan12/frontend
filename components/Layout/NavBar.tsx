import React, { ReactNode } from 'react'
import Link from 'next/link'
import { Flex, Image } from 'rebass'
import styled from 'styled-components'

declare interface NavBarInterface {
  items?: { text: string; href: string }[]
  children?: JSX.Element[] | JSX.Element
}

const LinksGroup = styled.div`
  a {
    margin-left: 2em;
    color: whitesmoke;
    font-size: 1.2em;
    text-decoration: none;
    &:hover {
      text-decoration: underline;
    }
  }
`

const StyledNav = styled(Flex)`
  padding: 0.5rem 3rem;
  background: ${({ theme }) => theme.colors.background};
  position: sticky;
  top: 0;
  left: 0;
`

const NavBar = ({
  items = [
    {
      text: 'Sign In',
      href: '/login'
    },
    {
      text: 'Sign Up',
      href: '/register'
    }
  ]
}: NavBarInterface) => {
  return (
    <StyledNav flexWrap="wrap" flexDirection="row" width={1} alignItems="center" justifyContent="space-between">
      <Link href="/">
        <a>
          <Image height="7vh" width="100%" src="../../static/logo.svg" />
        </a>
      </Link>
      <LinksGroup>
        {items.map((el: { href: string; text: ReactNode }) => (
          <Link href={el.href} key={el.href}>
            <a>{el.text}</a>
          </Link>
        ))}
      </LinksGroup>
    </StyledNav>
  )
}

export default NavBar
