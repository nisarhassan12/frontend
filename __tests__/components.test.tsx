import React from 'react'
import AppContext from '../lib/context'
import NavBar from '../components/Layout/NavBar'
import { create } from 'react-test-renderer'
import Logo from '../assets/logo'

describe('<NavBar />', () => {
  const sampleUser = {
    userId: 1,
    name: 'testUser',
    avatarUrl: ''
  }
  const initialContext = {
    isSignedIn: false,
    user: sampleUser
  }
  const loginLinks = [
    {
      href: '/auth',
      text: 'Sign In'
    },
    {
      href: '/register',
      text: 'Sign Up'
    }
  ]

  it('Navbar without Sign In renders correctly', () => {
    const tree = create(
      <AppContext.Provider value={initialContext}>
        <NavBar Logo={Logo} loginLinks={loginLinks} />
      </AppContext.Provider>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('Navbar with renders with Sign In too', () => {
    const tree = create(
      <AppContext.Provider value={{ ...initialContext, isSignedIn: true }}>
        <NavBar Logo={Logo} loginLinks={loginLinks} />
      </AppContext.Provider>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
