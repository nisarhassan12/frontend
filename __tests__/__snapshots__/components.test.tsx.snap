// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders Form with items 1`] = `
<form
  action="/"
  className="Form__StyledForm-sc-1qkwnfg-0 bGzgGo"
  method="POST"
  onSubmit={[Function]}
>
  <input
    name="username"
    placeholder="What is ur name?"
    type="password"
  />
</form>
`;

exports[`<Meta /> should render meta tags 1`] = `null`;

exports[`<NavBar /> renders NavBar with custom props 1`] = `
<div
  className="sc-bdVaJa sc-bwzfXH NavBar__StyledNav-sc-5bxc2y-1 jGuGMY"
  width={1}
>
  <a
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <img
      className="sc-bdVaJa sc-bZQynM ilartJ"
      height="7vh"
      src="../../static/logo.svg"
      width="100%"
    />
  </a>
  <div
    className="NavBar__LinksGroup-sc-5bxc2y-0 blbCBZ"
  >
    <a
      href="/profile"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      My profile
    </a>
  </div>
</div>
`;

exports[`<NavBar /> renders NavBar without props 1`] = `
<div
  className="sc-bdVaJa sc-bwzfXH NavBar__StyledNav-sc-5bxc2y-1 jGuGMY"
  width={1}
>
  <a
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <img
      className="sc-bdVaJa sc-bZQynM ilartJ"
      height="7vh"
      src="../../static/logo.svg"
      width="100%"
    />
  </a>
  <div
    className="NavBar__LinksGroup-sc-5bxc2y-0 blbCBZ"
  >
    <a
      href="/login"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Sign In
    </a>
    <a
      href="/register"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Sign Up
    </a>
  </div>
</div>
`;
